--- wiring.c	2019-03-01 14:57:00.000000000 -0500
+++ mywiring.c	2019-03-01 14:56:55.000000000 -0500
@@ -39,6 +39,9 @@
 volatile unsigned long timer0_millis = 0;
 static unsigned char timer0_fract = 0;
 
+// the log2 (number of millis) between context switches
+#define CONTEXT_SWITCH 3
+
 #if defined(TIM0_OVF_vect)
 ISR(TIM0_OVF_vect)
 #else
@@ -60,6 +63,9 @@
 	timer0_fract = f;
 	timer0_millis = m;
 	timer0_overflow_count++;
+	if (((m >> CONTEXT_SWITCH) << CONTEXT_SWITCH) == m) {
+	   yield ();
+        }
 }
 
 unsigned long millis()
